container/go/pkg/oci/reader.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Reads an OCI image layout on disk.
+// https://github.com/opencontainers/image-spec/blob/master/image-layout.md
+package oci
+
+import (
+	"fmt"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/pkg/errors"
+)
+
+// Read returns a docker image from the given path. The docker image should have been written by "Write".
+// The image in the given path <src> follows the OCI Image Layout.
+// (https://github.com/opencontainers/image-spec/blob/master/image-layout.md#oci-image-layout-specification)
+// Specifically, <src> must contains "index.json" that servers as an entrypoint for the contained image.
+// The image content and configs must be non-empty and stored in <src>/blobs/<SHAxxx>/.
+// NOTE: this only reads index with a single image.
+func Read(src string) (v1.Image, error) {
+	// Reading the image layout.
```

Comment:
Actually I would just remove this comment because I don't see it adding much value. The error handling right after gives me info to understand what's going on here.